panther_analysis_tool/main.py

```diff
@@ -888,38 +926,97 @@ def run_tests(
         # check expected result
         if result != unit_test["ExpectedResult"]:
             test_result["outcome"] = "FAIL"
+            failed_tests[analysis.get("PolicyID") or analysis["RuleID"]].append(unit_test["Name"])
+
+        # print results
+        print("\t[{}] {}".format(test_result["outcome"], unit_test["Name"]))
 
-        # check dedup and title function return non-None
+        # check custom function return non-None
         # Only applies to rules which match an incoming event
         if unit_test["ExpectedResult"]:
-            for func in ["dedup", "title"]:
-                if analysis_funcs.get(func):
-                    if not analysis_funcs[func](test_case):
-                        test_result[func] = "FAIL"
-                        test_result["outcome"] = "FAIL"
+            verify_result = _verify_test_run(test_case, mock_methods, analysis_funcs)
+            for function_name in verify_result:
+                if verify_result[function_name] != "PASS":
+                    test_result[function_name] = test_result["outcome"] = "FAIL"
```

Comment:
Yeah right here I think this should fail the test
```suggestion
                    test_result[function_name] = test_result["outcome"] = "FAIL"
                    failed_tests[analysis.get("PolicyID") or analysis["RuleID"]].append(unit_test["Name"])
```